### ISSUE TITLE
Python: InputImage.read_image signature now requires subimage and miplevel?

### ISSUE THREAD

**Issue by filmkorn** [OpenImageIO/oiio#2676]
**Describe the bug**
I'm wondering if there is a small mismatch between the docs and the signature of `InputImage.read_image`. [In the docs](https://openimageio.readthedocs.io/en/release-2.2.4.0-beta1/pythonbindings.html?highlight=AppendSubimage#ImageInput.read_image) the following signature is listed:

```python
ImageInput.read_image(chbegin, chend, format='float')
```

This signature is compatible with OIIO-1 but doesn't work anymore.

**To Reproduce**

```python
>>> image_input = ImageInput.create("path/to/some/file.exr")
>>> image_input.read_image(chbegin=1, chend=3, format=TypeHalf)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
TypeError: read_image(): incompatible function arguments. The following argument types are supported:
    1. (self: OpenImageIO.ImageInput, subimage: int, miplevel: int, chbegin: int, chend: int, format: OpenImageIO.TypeDesc = <TypeDesc 'float'>) -> object
    2. (self: OpenImageIO.ImageInput, format: OpenImageIO.TypeDesc = <TypeDesc 'float'>) -> object

Invoked with: <OpenImageIO.ImageInput object at 0x7f02140cab90>, kwargs: chbegin=1, chend=3, format=<TypeDesc 'half'>
```

The signature from the docs is not listed here.

**Expected behavior**
Not sure - no TypeError is raised or the docs may need an update?

**Evidence**

See above.

**Platform information:**
 - OIIO branch/version: 2.2.4.0-beta1
 - OS: Cent7
 - C++ compiler: gcc-6.4
 - Any non-default build flags when you build OIIO:

**Comment by lgritz** [comment]
You are correct. Fixing..

**Comment by lgritz** [comment]
Proposed fix #2677

**Comment by filmkorn** [comment]
@lgritz Thank you!
